VirtoCommerce.Platform.Tests/UnitTests/FileVersionProviderTest.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Mvc.TagHelpers.Internal;
+using Microsoft.AspNetCore.WebUtilities;
+using Xunit;
+using FileVersionProvider = VirtoCommerce.Platform.Core.VersionProvider.FileVersionProvider;
+
+namespace VirtoCommerce.Platform.Tests.UnitTests
+{
+    public class FileVersionProviderTest
+    {
+        [Fact]
+        public void TestVersionAppend()
+        {
+            const string fileName = "test.txt";
+
+            var path = CreateTemporaryFile(fileName);
+
+            var versionProvider = new FileVersionProvider();
+
+            var actualHash = versionProvider.GetFileVersion(Path.Join(path, fileName));
+
+            var expectedHash = GetFileHash(Path.Join(path, fileName));
+
+            RemoveDirectoryAndFile(path, fileName);
+
+            Assert.Equal(expectedHash, actualHash);
+        }
+
+        [Fact]
+        public void TestFileNotExists()
+        {
+            var versionProvider = new FileVersionProvider();
+
+            var actual = versionProvider.GetFileVersion("wrongPath/test.txt");
+
+            Assert.Null(actual);
+        }
+
+        [Fact(Skip = "TODO")]
+        public void TestCaching()
+        {
+
+        }
+
+        private string CreateTemporaryFile(string fileName)
+        {
+            var tempPath = Path.GetTempPath();
+
+            var fullPhysicalPath = Path.Join(tempPath, Guid.NewGuid().ToString());
+
+            Directory.CreateDirectory(fullPhysicalPath);
+
+            File.Create(Path.Join(fullPhysicalPath, fileName)).Dispose();
+
+            return fullPhysicalPath;
```

Comment:
You can just use the [`Path.GetTempFileName()`](https://docs.microsoft.com/ru-ru/dotnet/api/system.io.path.gettempfilename?view=netframework-4.7.2) method instead.